tonic-examples/src/tls_client_auth/server.rs

```diff
@@ -0,0 +1,74 @@
+pub mod pb {
+    tonic::include_proto!("grpc.examples.echo");
+}
+
+use std::collections::VecDeque;
+
+use pb::{EchoRequest, EchoResponse};
+use tonic::transport::{Certificate, Identity, Server, ServerTlsConfig};
+use tonic::{Request, Response, Status, Streaming};
+
+type EchoResult<T> = Result<Response<T>, Status>;
+type Stream = VecDeque<Result<EchoResponse, Status>>;
+
+#[derive(Default)]
+pub struct EchoServer;
+
+#[tonic::async_trait]
+impl pb::server::Echo for EchoServer {
+    async fn unary_echo(&self, request: Request<EchoRequest>) -> EchoResult<EchoResponse> {
+        let message = request.into_inner().message;
+        Ok(Response::new(EchoResponse { message }))
+    }
+
+    type ServerStreamingEchoStream = Stream;
+
+    async fn server_streaming_echo(
```

Comment:
btw to simplify this you can just drop all these unimplemented since codegen now provides a default impl that does this.

Reply:
Oh, good to know - I'll do that.

Reply:
I've pushed a new commit which removes the unnecessary implementations. It does seem that the associated types still need to be defined, however?

Reply:
Yeah, we don't have default associated types yet unfortunetly